app/Http/Controllers/Api/SubjectController.php

```diff
@@ -12,9 +12,9 @@
 
 class SubjectController extends Controller
 {
-    public function index()
+    public function index(Repository $repository)
     {
-    	return Subjects::collection(Subject::get());
+    	return Subjects::collection($repository->getAll());
```

Comment:
Subject::collection() digunakan untuk memanggil resource untuk collection of records. Cara lain adalah membuat Resource SubjectsCollection.